src/megaclient.cpp

```diff
@@ -13024,203 +12946,152 @@ error MegaClient::isLocalPathSyncable(string newPath, int newSyncTag, SyncError
+    error e = API_OK;
+    syncs.forEachSyncConfig([&](const SyncConfig& config){
 
-        if (config.getEnabled() && !isAnError(config.getError()) &&
-                ( newLocallyEncodedAbsolutePath.isContainingPathOf(otherLocallyEncodedAbsolutePath)
-                  || otherLocallyEncodedAbsolutePath.isContainingPathOf(newLocallyEncodedAbsolutePath)
-                ) )
+        if (config.getTag() != newSyncTag)
         {
+            LocalPath otherLocallyEncodedPath = LocalPath::fromPath(config.getLocalPath(), *fsaccess);
+            LocalPath otherLocallyEncodedAbsolutePath;
+            fsaccess->expanselocalpath(otherLocallyEncodedPath, otherLocallyEncodedAbsolutePath);
 
-            if (syncError)
+            if (config.getEnabled() && !config.getError() &&
+                    ( newLocallyEncodedAbsolutePath.isContainingPathOf(otherLocallyEncodedAbsolutePath)
+                      || otherLocallyEncodedAbsolutePath.isContainingPathOf(newLocallyEncodedAbsolutePath)
+                    ) )
             {
-                *syncError = LOCAL_PATH_SYNC_COLLISION;
+
+                if (syncError)
+                {
+                    *syncError = LOCAL_PATH_SYNC_COLLISION;
+                }
+                e = API_EARGS;
             }
-            return API_EARGS;
         }
-    }
+    });
 
-    return API_OK;
+    return e;
 }
 
 // check sync path, add sync if folder
 // disallow nested syncs (there is only one LocalNode pointer per node)
 // (FIXME: perform the same check for local paths!)
-error MegaClient::addsync(SyncConfig syncConfig, const char* debris, LocalPath* localdebris, SyncError &syncError, bool delayInitialScan, void *appData)
+error MegaClient::checkSyncConfig(SyncConfig& syncConfig, LocalPath& rootpath, std::unique_ptr<FileAccess>& openedLocalFolder, Node*& remotenode, bool& inshare, bool& isnetwork)
 {
 #ifdef ENABLE_SYNC
-    syncError = NO_SYNC_ERROR;
 
-    Node* remotenode = nodebyhandle(syncConfig.getRemoteNode());
-    bool inshare = false;
+    // Checking for conditions where we would not even add the sync config
+    // Though, if the config is already present but now invalid for one of these reasonse, we don't remove it
+
+    syncConfig.mEnabled = true;
+    syncConfig.mError = NO_SYNC_ERROR;
+    syncConfig.mWarning = NO_SYNC_WARNING;
+
+    remotenode = nodebyhandle(syncConfig.getRemoteNode());
+    inshare = false;
     if (!remotenode)
     {
-        LOG_err << "Sync root does not exist in the cloud: "
-                << syncConfig.getLocalPath()
-                << ": "
-                << LOG_NODEHANDLE(syncConfig.getRemoteNode());
+        LOG_warn << "Sync root does not exist in the cloud: "
+                 << syncConfig.getLocalPath()
+                 << ": "
+                 << LOG_NODEHANDLE(syncConfig.getRemoteNode());
 
-        syncError = REMOTE_NODE_NOT_FOUND;
+        syncConfig.mError = REMOTE_NODE_NOT_FOUND;
```

Comment:
This function checkSyncConfig() is called from two places - addsync, in which case the sync is not even added to our set of syncs yet (so can't be disabled), and also enablesync(), where it is in the set of syncs, but not running.  We do notify the app for the second case (if the error is different to one previously reported), but it wouldn't make sense for the first case.    This one also goes back to the possibility of separating out checkSyncConfig() for clearer code.

Reply:
My suggestion was more in the direction of moving the `mEnabled = false` triggered from the usage of `checkSyncConfig()` at `UnifiedSync::enableSync()`. Also, take into account that for the rest of errors managed in `checkSyncConfig()`, the method already sets the config as disabled (I should have said "we should set the config as disabled", instead of "we should disable the sync").
I still think my suggestion is appropriate, without requiring to separate out `checkSyncConfig()`.

Reply:
Oh, I see what you mean, thanks for the further explanation.  yes I think that's fine for both use cases - from addsync() it doesn't matter, the config object is discarded as it was never added, and for the enableSync case it keeps the setting of error and enable flag together.  I'll make the change.

Reply:
While making that update, I noticed addsync() always called enableSync(), even for configs that have mEnabled==false.   I've also added a check there that allows for adding syncs in a disabled state (ie, skip the enableSync() in that case)